lib/Runtime/Library/JavascriptBigInt.h

```diff
@@ -0,0 +1,69 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+#pragma once
+
+namespace Js
+{
+    class JavascriptBigInt sealed : public RecyclableObject
+    {
+
+    private:
+        Field(BOOL) m_isNegative;
+        Field(uint32*) m_digits;         // digits
+        Field(uint32) m_length;          // length
+
+        // TODO: BigInt should be arbitrary-precision, need to resize space and add tagged BigInt
+        static const uint32 MaxStringLength = 100; // Max String length
+        static const uint32 MaxDigitLength = 20;  // Max Digit length 
+        uint32 m_reservedDigitsInit[MaxDigitLength]; // pre-defined space to store array
+
+        DEFINE_VTABLE_CTOR(JavascriptBigInt, RecyclableObject);
+
+    public:
+        JavascriptBigInt(StaticType * type)
+            : RecyclableObject(type), m_length(0), m_digits(nullptr)
+        {
+            Assert(type->GetTypeId() == TypeIds_BigInt);
+            m_isNegative = false;
+        }
+
+        bool MulThenAdd(uint32 luMul, uint32 luAdd);
```

Comment:
I imagine this is probably for a future change, but I think that if you find and replace all of these `uint32` with `intptr_t` then we can pretty easily have each platform operating on its natural data size.